libraries/AP_Logger/AP_Logger_Block.cpp

```diff
@@ -822,10 +822,13 @@ bool AP_Logger_Block::logging_failed() const
     return false;
 }
 
+// detect whether the IO thread is running, since this is considered a catastrophic failure for the logging system
+// better be really, really sure
 bool AP_Logger_Block::io_thread_alive() const
 {
-    // if the io thread hasn't had a heartbeat in 1s it is dead
-    return (AP_HAL::millis() - io_timer_heartbeat) < 1000U;
+    uint32_t now = AP_HAL::millis();
+    // if the io thread hasn't had a heartbeat in 1s and its some time after startup it is dead
+    return now > 10000U && (now - io_timer_heartbeat) < 1000U;
```

Comment:
This is always false for the first 10 seconds. Is that what you want?
It seams to go against the explanation on the PR.

Reply:
Yeah I just realized - now corrected